src/Data/List/Relation/Unary/Sorted/TotalOrder/Properties.agda

```diff
@@ -109,22 +113,30 @@ module _ (DO : DecTotalOrder a ℓ₁ ℓ₂) where
                 Connected _≤_ (just v) (head xs) →
                 Connected _≤_ (just v) (head ys) →
                 Connected _≤_ (just v) (head (merge _≤?_ xs ys))
-    merge-con {xs = []}     {[]}     cxs cys = cys
-    merge-con {xs = []}     {y ∷ ys} cxs cys = cys
+    merge-con {xs = []}              cxs cys = cys
     merge-con {xs = x ∷ xs} {[]}     cxs cys = cxs
     merge-con {xs = x ∷ xs} {y ∷ ys} cxs cys with x ≤? y
     ... | yes x≤y = cxs
     ... | no  x≰y = cys
 
   merge⁺ : ∀ {xs ys} → Sorted O xs → Sorted O ys → Sorted O (merge _≤?_ xs ys)
-  merge⁺ {[]}     {[]}     rxs rys = []
-  merge⁺ {[]}     {x ∷ ys} rxs rys = rys
+  merge⁺ {[]}              rxs rys = rys
   merge⁺ {x ∷ xs} {[]}     rxs rys = rxs
-  merge⁺ {x ∷ xs} {y ∷ ys} rxs rys with x ≤? y |
-    merge⁺ (Linked.tail rxs) rys | merge⁺ rxs (Linked.tail rys)
+  merge⁺ {x ∷ xs} {y ∷ ys} rxs rys
+   with x ≤? y  | merge⁺ (Linked.tail rxs) rys
+                      | merge⁺ rxs (Linked.tail rys)
   ... | yes x≤y | rec | _   = merge-con (head′ rxs)      (just x≤y)  ∷′ rec
   ... | no  x≰y | _   | rec = merge-con (just (≰⇒≥ x≰y)) (head′ rys) ∷′ rec
 
+  -- Reexport merge-is-superlistˡʳ
+
+  open
+    SublistProperties.Merge
+      (Preorder.Eq.setoid (DecTotalOrder.preorder DO))
+       _≤_
+      (DecTotalOrder.isDecTotalOrder DO)
+    public using (merge-is-superlistˡ; merge-is-superlistʳ)
```

Comment:
Hmm although this style is tempting, we try and avoid writing it where possible as it makes it very confusing for users to work out what the final type actually is. I think it's better to write out the final type, and then apply the more general lemma directly (e.g. see `Sorted⇒AllPairs` above)

Reply:
Ok, fixed this.